### PR TITLE
Add tests to verify analyzer configuration with analysis properties 

### PR DESCRIPTION
**Pull request by mavasani**
Tests added:

1. Add test to verify mapping from analysis properties to analyzer config file
2. Add test to verify bulk escalation of CA rules with different analysis modes

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @jmarolf @Youssef1313 for review

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToFloatWarningLevels.cs [hunk 1]
```diff
@@ -259,5 +260,95 @@ static void Main()
...
+        [InlineData("5", "", "true")]
+        [InlineData("5.0", "minimum", "false")]
+        [InlineData("6", "recommended", "false")]
+        [InlineData("6.0", "", "true")]
+        [InlineData("7", "none", "true")]
+        [InlineData("7.0", "", "false")]
+        [InlineData("8", "default", "false")]
+        [InlineData("8.0", "", "true")]
+        [RequiresMSBuildVersionTheory("16.8")]
+        public void It_maps_analysis_properties_to_globalconfig(string analysisLevel, string analysisMode, string codeAnalysisTreatWarningsAsErrors)
```
Comment (baronfel) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToFloatWarningLevels.cs:274]: To make sure I understand - there is an implicit dependency between roslyn and SDK that we didn't quite get to when we bumped the TFM to net8.0, and this test is making the dependency explicit so we don't later this year when we start the road to 9.0?

Comment (mavasani) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToFloatWarningLevels.cs:274]: Yes, that is correct.

The NetAnalyzers NuGet package drops props and targets file and some analyzerconfig files along with the analyzer assemblies. These props and targets file have the functionality to map MSBuild properties being tested here to a specific analyzerConfig file that is then passed to the compiler command line for appropriate analyzer configuration. We need to ensure that whenever LatestAnalysisLevel is bumped to 9.0 in future, the tooling that generates and packs these analyzer config files into the repo should be aware of the version bump to produce 9.0 specific analyzer config files.

Comment (baronfel) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToFloatWarningLevels.cs:274]: Sounds perfect - love that we're canonicalizing all of these touchpoints with tests!

**Review by baronfel — APPROVED** [review]

**Comment by baronfel** [comment]
Happy to take this whenever, figured that @jmarolf might want to see it too.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToFloatWarningLevels.cs (hunk 1 above)
Comment (mavasani) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToFloatWarningLevels.cs:274]: Yes, I am planning to add a few more of these over the coming days.

**Comment by mavasani** [comment]
@baronfel I added another test to the PR, can you please take a look? Please feel free to merge the PR once you are ready to approve. @jmarolf and me have discussed a bunch in the past about the need for these tests, so we should be good to add these.

**Comment by baronfel** [comment]
Great stuff @mavasani - thanks for taking the time.

**Pull request merged**